diff --git a/src/calibre/devices/smart_device_app/driver.py b/src/calibre/devices/smart_device_app/driver.py
index 87608f305b..163596c9d6 100644
--- a/src/calibre/devices/smart_device_app/driver.py
+++ b/src/calibre/devices/smart_device_app/driver.py
@@ -2175,7 +2175,7 @@ def monkeypatch_zeroconf():
     # "monkeypatch" zeroconf with a function without the check
     try:
         from zeroconf._utils.name import service_type_name
-        service_type_name.__kwdefaults__['strict'] = False
+        # service_type_name.__kwdefaults__['strict'] = False
     except ImportError:
         import zeroconf
         zeroconf.service_type_name = monkeypatched_service_type_name
